File: front/src/store/modules/studyRecord.js
import axios from "axios";

const state = {
  studyRecords: [],
  studyRecord: {
    id: '',
    user_id: '',
    task_id: '',
    study_material_id: '',
    start_time: '',
    end_time: '',
    is_finished: '',
    created_at: '',                
    updated_at: '',
  }
};

const getters = {
  studyRecords: state => state.studyRecords
}

const mutations = {

};

const action = {

}

export default {
  state,
  getters,
  mutations,
  actions,
}

File: back/app/models/study_record.rb
class StudyRecord < ApplicationRecord
  include TaskBase
  belongs_to :user
  belongs_to :task
  belongs_to :study_material

  validates :user_id, presence: true
  validates :task_id, presence: true
  validates :study_material_id, presence: true
  validates :start_time, presence: true
  validates :end_time, presence: true

end
